pravega/pravega
# Issue 4247:  Remove lombok and netty-all from client's dependency list.

**Change log description**  
- Ensure lombok is not part of the client's dependency list.
- Remove redundant netty-all dependency from `:shared:controller-api`

**Purpose of the change**  
Reduces dependencies. Fixes #4247.

**What the code does**  
- Removing `lombok` and `netty-all` reduces the dependencies of the client by 5+MB.
- Ensure `javadoc` gradle task runs on the delombok'ed code.
- Update gradle lombok plugin to `3.2.0`

**How to verify it**  
Ensure java docs of lombok annotated code are generated and ensure `:client: dependencies` does not list lombok and netty-all as part of its `runtimeOnly` dependencies.
